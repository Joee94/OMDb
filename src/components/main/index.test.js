import * as React from 'react';
import renderer from 'react-test-renderer';
import Main from './index';

jest.mock('./cards', () => () => <mock-Cards />);

describe('Main', () => {
	describe('when not loading', () => {
		const contextValues = { state: { response: { loading: false } } };
		jest.spyOn(React, 'useContext').mockImplementation(() => contextValues);
		const component = renderer.create(<Main />);
		it('should create the Main component', () => {
			expect(component).toMatchSnapshot();
		});
	});
	describe('when loading', () => {
		const contextValues = { state: { response: { loading: true } } };
		jest.spyOn(React, 'useContext').mockImplementation(() => contextValues);
		const component = renderer.create(<Main />);
		it('should create the Main component', () => {
			expect(component).toMatchSnapshot();
		});
	});
});
