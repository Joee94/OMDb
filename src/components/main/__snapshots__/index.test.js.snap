// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main when loading should create the Main component 1`] = `
<main>
  <object
    data="loading.svg"
    type="image/svg+xml"
  >
    svg-animation
  </object>
  <mock-Cards />
</main>
`;

exports[`Main when not loading should create the Main component 1`] = `
<main>
  <mock-Cards />
</main>
`;
